core/src/main/java/jenkins/install/InstallUtil.java

```diff
@@ -204,6 +208,21 @@ private static InstallState getDefaultInstallState() {
         }
     }
 
+    private static boolean isEdgeCaseUpgrade() {
+        Jenkins j = Jenkins.getInstance();
+        // Edge case: used Jenkins 1 but did not save the system config page,
+        // the version is not persisted and returns 1.0, so try to check if
+        // they actually did anything
+        Map<String, TopLevelItem> items = new HashMap<String, TopLevelItem>(j.getItemMap());
+        for (Iterator<Map.Entry<String, TopLevelItem>> it = items.entrySet().iterator(); it.hasNext();) {
+            Map.Entry<String, TopLevelItem> entry = it.next();
+            if (new File(entry.getValue().getRootDir(), BUILT_IN_MARKER_FILE).exists()) {
```

Comment:
How do I create this `.builtIn` file on the file system of the Jenkins Home if I have create my "TopLevelItem" with a groovy script? For example in an init.groovy script?